src/cli/subcommand/install.rs

```diff
@@ -43,28 +46,66 @@ impl CommandExecute for Install {
             explain,
         } = self;
 
-        let mut plan = match &plan {
-            Some(plan_path) => {
-                let install_plan_string = tokio::fs::read_to_string(&plan_path)
+        let existing_receipt: Option<InstallPlan> = match Path::new(RECEIPT_LOCATION).exists() {
+            true => {
+                let install_plan_string = tokio::fs::read_to_string(&RECEIPT_LOCATION)
                     .await
                     .wrap_err("Reading plan")?;
+                Some(serde_json::from_str(&install_plan_string)?)
+            },
+            false => None,
+        };
+
+        let mut install_plan = match (planner, plan) {
+            (Some(planner), None) => {
+                let chosen_planner: Box<dyn Planner> = planner.clone().boxed();
+
+                match existing_receipt {
+                    Some(existing_receipt) => {
+                        if existing_receipt.planner.typetag_name() != chosen_planner.typetag_name() {
+                            return Err(eyre!("Found existing plan in `/nix/receipt.json` which used a different planner, try uninstalling the existing install"))
+                        }
+                        if existing_receipt.planner.settings().map_err(|e| eyre!(e))? != chosen_planner.settings().map_err(|e| eyre!(e))? {
+                            return Err(eyre!("Found existing plan in `/nix/receipt.json` which used different planner settings, try uninstalling the existing install"))
+                        }
+                        if existing_receipt.actions.iter().all(|v| v.action_state() == ActionState::Completed) {
+                            return Err(eyre!("Found existing plan in `/nix/receipt.json`, with the same settings, already completed, try uninstalling and reinstalling if Nix isn't working"))
+                        }
```

Comment:
It appears there's a constant RECEIPT_LOCATION -- can we use that in these messages?

Reply:
Oh heck yeah good idea